fedimint-core/src/tiered_multi.rs

```diff
@@ -82,14 +84,14 @@ impl<T> TieredMulti<T> {
 
     /// Verifies whether the structure of `self` and `other` is identical
     pub fn structural_eq<O>(&self, other: &TieredMulti<O>) -> bool {
-        let tier_eq = self.0.keys().eq(other.0.keys());
-        let per_tier_eq = self
-            .0
-            .values()
-            .zip(other.0.values())
-            .all(|(c1, c2)| c1.len() == c2.len());
-
-        tier_eq && per_tier_eq
+        let tier_eq = || self.0.keys().eq(other.0.keys());
```

Comment:
Oh. Lambdas again. You really seem to like lambdas. :)

Is it in hopes of making `per_tier_eq` lazily evaluated in the `&&`? Wouldn't compiler probably be able to see through this expression having no side-effects and optimize anyway? Hmmm... maybe... maybe not ...

Anyway - if there's a good reason, please add a comment, otherwise someone is going to revert it back eventually.